quesoqueue.cpp

```diff
@@ -222,6 +222,41 @@ std::optional<Level> QuesoQueue::Current() {
     return _current;
 }
 
+std::optional<Level> QuesoQueue::Random() {
+    auto list = List();
+    std::deque<Level> levels(std::get<0>(list)); // Use online levels first.
+
+    if (levels.empty()) {
+        levels = std::get<1>(list); // See if there are offline levels.
+        if (levels.empty()) {
+            return std::nullopt;
+        }
+    }
+    std::vector<Level> new_current = {};
```

Comment:
Why is current in a vector?

Reply:
I guess it's because you're trying to use std::sample. This seems weird, why not just generate an index and not use the extra vector? This leads to weird stuff like "new_current[0]"

Reply:
Eh, what can I say?  I like using these standard functions for things where I can accidentally insert bugs without realizing (https://stackoverflow.com/a/6943003).